### PR TITLE
Internal 79 dgraph disk monitoring

### PR DESCRIPTION
**Pull request by jgrillo-grapl**
### Which issue does this PR correspond to?

https://github.com/grapl-security/issue-tracker/issues/79

### What changes does this PR make to Grapl? Why?

This PR changes how we create a Docker Swarm cluster for running DGraph, adds monitoring to the DGraph instances, and configures CloudWatch Alarms for disk usage on the DGraph instances. Previously we were using `docker-machine` to create AWS instance to deploy our Docker Swarm cluster on. However, `docker-machine` apparently has been deprecated: https://github.com/docker/machine/issues/4537. Using regular AWS resources via CDK (e.g. an Autoscaling Group) seems more sustainable. One additional benefit to these changes is that we can block SSH access from the DGraph instances and use SSM instead.

### How were these changes tested?

These changes were tested manually in an AWS sandbox account.

### CODE REVIEW

File: src/js/grapl-cdk/bin/grapl-cdk.ts
Comment (jgrillo-grapl) [inline src/js/grapl-cdk/bin/grapl-cdk.ts:13]: oops

File: src/js/grapl-cdk/lib/grapl-cdk-stack.ts
Comment (jgrillo-grapl) [inline src/js/grapl-cdk/lib/grapl-cdk-stack.ts:479]: This isn't a thing anymore, will have to re-work how we create these alarms.

File: src/js/grapl-cdk/lib/swarm.ts
Comment (inickles-grapl) [inline src/js/grapl-cdk/lib/swarm.ts:108]: Yeah, this looks like it could read logs from other deployments in the account.

I'm not sure what the logs arns look like, but hopefully the resource names begin with deployment name (or can) so we could limit to a deployment.

**Review by inickles-grapl — CHANGES_REQUESTED** [review]
A couple change requests, one question and looks like deployment values got checked in.

File: src/js/grapl-cdk/lib/swarm.ts
Comment (inickles-grapl) [reply src/js/grapl-cdk/lib/swarm.ts:108]: Oh, I guess just overwrite, not read, so not as bad.

Comment (jgrillo-grapl) [reply src/js/grapl-cdk/lib/swarm.ts:109]: I think I should be able to lock this down to a single log group's ARN.

Comment (jgrillo-grapl) [reply src/js/grapl-cdk/lib/swarm.ts:109]: derp nvm this does things with log streams too.. still think I should be able to lock it down more, investigating.

File: src/js/grapl-cdk/lib/grapl-cdk-stack.ts
Comment (jgrillo-grapl) [reply src/js/grapl-cdk/lib/grapl-cdk-stack.ts]: done

File: src/js/grapl-cdk/lib/swarm.ts
Comment (jgrillo-grapl) [reply src/js/grapl-cdk/lib/swarm.ts]: done

File: src/js/grapl-cdk/lib/grapl-cdk-stack.ts
Comment (jgrillo-grapl) [reply src/js/grapl-cdk/lib/grapl-cdk-stack.ts]: done

File: src/js/grapl-cdk/bin/deployment_parameters.ts
Comment (jgrillo-grapl) [reply src/js/grapl-cdk/bin/deployment_parameters.ts]: done

File: src/js/grapl-cdk/bin/deployment_parameters.ts
Comment (jgrillo-grapl) [reply src/js/grapl-cdk/bin/deployment_parameters.ts]: fixed

File: src/js/grapl-cdk/README.md
Comment (wimax-grapl) [inline src/js/grapl-cdk/README.md:37]: I'll be changing this after I land my change to that script :)

File: src/js/grapl-cdk/README.md
Comment (wimax-grapl) [inline src/js/grapl-cdk/README.md:170]: <3 <3

File: src/js/grapl-cdk/lib/grapl-cdk-stack.ts
Comment (wimax-grapl) [inline src/js/grapl-cdk/lib/grapl-cdk-stack.ts:532]: one thing to check:
when i was dealing with sagemaker arn stuff, I actually had to lower-case-ify the resourceName; you may want to check if the same restriction/expectation is the case here (in case the prefix has upper case). I noticed you have to lowercase the bucketname as well 🤔

File: src/js/grapl-cdk/lib/grapl-cdk-stack.ts
Comment (wimax-grapl) [inline src/js/grapl-cdk/lib/grapl-cdk-stack.ts:855]: thanks to the above
`export const graplVersion`
i don't think we need to do this default anymore; and it looks like `.version` is not `?` anyway

File: src/js/grapl-cdk/lib/service.ts
Comment (wimax-grapl) [inline src/js/grapl-cdk/lib/service.ts:116]: not a blocking request, but a nice-to-have: 
could you add something to the issue tracker, and cross-reference it here?

File: src/js/grapl-cdk/swarm/swarm_setup.py
Comment (wimax-grapl) [inline src/js/grapl-cdk/swarm/swarm_setup.py:48]: could we do either a print("Checking status...") or an eventual-timeout here?

i find that "while true: silently try something, then sleep, forever and ever" are hard to debug for consumers, in case something's genuinely fucked up

it's the root of a thing that screwed up andrea the other day
https://github.com/grapl-security/grapl/pull/429/files#diff-07bf86199f1865f81c28ab7cdb4d0b3572d6b36587479b3a617b6ce066fe01beL53-L69

File: src/python/swarm-lifecycle-event-handler/app.py
Comment (wimax-grapl) [inline src/python/swarm-lifecycle-event-handler/app.py:134]: don't super love the duck typing based on "does this substring appear in json" tbh
i think for a lot of these you can just do like
```
notif_meta = message.get("NotificationMetadata", None)
if notif_meta:
```

**Review by wimax-grapl — APPROVED** [review]
Approved, but I'd really like it if you'd fix the final comment (ctrl-f duck typing)

**Review by inickles-grapl — APPROVED** [review]

File: src/js/grapl-cdk/lib/service.ts
Comment (jgrillo-grapl) [reply src/js/grapl-cdk/lib/service.ts:116]: https://github.com/grapl-security/issue-tracker/issues/106

File: src/js/grapl-cdk/lib/grapl-cdk-stack.ts
Comment (jgrillo-grapl) [reply src/js/grapl-cdk/lib/grapl-cdk-stack.ts]: fixed

File: src/js/grapl-cdk/lib/grapl-cdk-stack.ts
Comment (jgrillo-grapl) [reply src/js/grapl-cdk/lib/grapl-cdk-stack.ts]: fixed

File: src/js/grapl-cdk/swarm/swarm_setup.py
Comment (jgrillo-grapl) [reply src/js/grapl-cdk/swarm/swarm_setup.py]: done

**Pull request merged**
